cooltools/expected.py

```diff
@@ -417,77 +499,48 @@ def trans_expected(clr, chromosomes, chunksize=1000000, use_dask=False):
-        x = [clr.extent(chrom)[1] - clr.extent(chrom)[0] for chrom in chromosomes]
-        pairblock_list = []
-        for i in range(n):
-            for j in range(i + 1, n):
-                # appending to the list of tuples
-                pairblock_list.append((chromosomes[i], chromosomes[j], x[i] * x[j]))
-        return pd.DataFrame(pairblock_list, columns=["chrom1", "chrom2", "n_total"])
-
-    def n_bad_trans_elements(clr, chromosomes):
-        n = 0
-        # bad bins are ones with
-        # the weight vector being NaN:
-        x = [
-            np.sum(clr.bins()["weight"].fetch(chrom).isnull().astype(int).values)
-            for chrom in chromosomes
-        ]
-        pairblock_list = []
-        for i in range(len(x)):
-            for j in range(i + 1, len(x)):
-                # appending to the list of tuples
-                pairblock_list.append((chromosomes[i], chromosomes[j], x[i] * x[j]))
-        return pd.DataFrame(pairblock_list, columns=["chrom1", "chrom2", "n_bad"])
+    warnings.warn(
+        "`cooltools.expected.trans_expected()` is deprecated in 0.3.2, will be removed subsequently. "
+        "Use `cooltools.expected.blocksum_pairwise()` instead.",
+        category=FutureWarning,
+        stacklevel=2,
+    )
 
     if use_dask:
         # pixels = daskify(clr.filename, clr.root + '/pixels', chunksize=chunksize)
         raise NotImplementedError("To be implemented once dask supports MultiIndex")
-    else:
-        pixels = clr.pixels()[:]
-    # getting pixels that belong to trans-area,
-    # defined by the list of chromosomes:
-    pixels = cooler.annotate(pixels, clr.bins(), replace=False)
-    pixels = pixels[
-        (pixels.chrom1.isin(chromosomes))
-        & (pixels.chrom2.isin(chromosomes))
-        & (pixels.chrom1 != pixels.chrom2)
-    ]
-    pixels["balanced"] = pixels["count"] * pixels["weight1"] * pixels["weight2"]
-    ntot = (
-        n_total_trans_elements(clr, chromosomes)
-        .groupby(("chrom1", "chrom2"))["n_total"]
-        .sum()
-    )
-    nbad = (
-        n_bad_trans_elements(clr, chromosomes)
-        .groupby(("chrom1", "chrom2"))["n_bad"]
-        .sum()
-    )
-    trans_area = ntot - nbad
-    trans_area.name = "n_valid"
-    # processing with use_dask=True is different:
-    if use_dask:
-        # trans_sum = pixels.groupby(['chrom1', 'chrom2'])['balanced'].sum().compute()
-        pass
-    else:
-        trans_sum = pixels.groupby(["chrom1", "chrom2"])["balanced"].sum()
-    # for consistency with the cis_expected function:
-    trans_sum.name = trans_sum.name + ".sum"
-
-    # returning a DataFrame with MultiIndex, that stores
-    # pairs of 'balanced.sum' and 'n_valid' values for each
-    # pair of chromosomes.
-    dtable = pd.merge(
-        trans_sum.to_frame(), trans_area.to_frame(), left_index=True, right_index=True
-    )
+
+    # turn chromosomes into supports:
+    chrom_supports = [ (chrom, 0, None) for chrom in chromosomes ]
```

Comment:
@nvictus I'm using this feature of `lib.common.assign_supports`: https://github.com/mirnylab/cooltools/blob/9c8028f756865f8315096419b3d32bc7d6560bed/cooltools/lib/common.py#L34

the feature is that `(chrom,0,None)` implies the entire chromosome , not sure is `clr.chromsizes[chrom]` would be better instead of `None`